static/src/javascripts/projects/common/modules/onward/history.js

```diff
@@ -0,0 +1,477 @@
+// @flow
+/* eslint-disable no-useless-escape */
+/*
+ Module: history.js
+ Description: Gets and sets users reading history
+ */
+import fastdom from 'fastdom';
+import $ from 'lib/$';
+import { local } from 'lib/storage';
+import { getPath } from 'lib/url';
+import isObject from 'lodash/objects/isObject';
+import pick from 'lodash/objects/pick';
+import mapValues from 'lodash/objects/mapValues';
+
+import type { bonzo } from 'bonzo';
+
+const editions = ['uk', 'us', 'au'];
+
+const editionalised = [
+    'business',
+    'commentisfree',
+    'culture',
+    'environment',
+    'media',
+    'money',
+    'sport',
+    'technology',
+];
+
+const pageMeta = [
+    {
+        tid: 'section',
+        tname: 'sectionName',
+    },
+    {
+        tid: 'keywordIds',
+        tname: 'keywords',
+    },
+    {
+        tid: 'seriesId',
+        tname: 'series',
+    },
+    {
+        tid: 'authorIds',
+        tname: 'author',
+    },
+];
+
+const buckets = [
+    {
+        type: 'content',
+        indexInRecord: 1,
+    },
+    {
+        type: 'front',
+        indexInRecord: 2,
+    },
+];
+
+const summaryPeriodDays = 90;
+const forgetUniquesAfter = 10;
+const historySize = 50;
+const storageKeyHistory = 'gu.history';
+const storageKeySummary = 'gu.history.summary';
+
+const today = Math.floor(Date.now() / 86400000); // 1 day in ms
+
+let historyCache;
+let summaryCache;
+let popularFilteredCache;
+let topNavItemsCache;
+let inMegaNav = false;
+
+const saveHistory = (history: Array<Array<any>>): void => {
+    historyCache = history;
+    local.set(storageKeyHistory, history);
+};
+
+const saveSummary = (summary: Object): void => {
+    summaryCache = summary;
+    local.set(storageKeySummary, summary);
+};
+
+const getHistory = (): Array<Array<any>> => {
+    historyCache = historyCache || local.get(storageKeyHistory) || [];
+    return historyCache;
+};
+
+const getSummary = (): Object => {
+    if (!summaryCache) {
+        summaryCache = local.get(storageKeySummary);
+
+        if (
+            !isObject(summaryCache) ||
+            !isObject(summaryCache.tags) ||
+            typeof summaryCache.periodEnd !== 'number'
+        ) {
+            summaryCache = {
+                periodEnd: today,
+                tags: {},
+                showInMegaNav: true,
+            };
+        }
+    }
+    return summaryCache;
+};
+
+const seriesSummary = (): Object => {
+    const views = item => item.reduce((acc, record) => acc + record[1], 0);
+
+    const seriesTags = pick(getSummary().tags, (v, k) => k.includes('series'));
+
+    const seriesTagsSummary = mapValues(
```

Comment:
Yes why not! This would do the same as mapValues...

```
const seriesTagsSummary = Object.keys(seriesTags).reduce((acc, val, index) => {
        const tag = seriesTags[val];
        acc[val] = views(tag[1]) + views(tag[2]);
       return acc;
    }, {});
```